tests/objcthemis/objthemis/SecureMessageTests.m

```diff
@@ -37,55 +37,105 @@ - (void)testGeneratingKeys {
+  [self signVerifyWithPrivateKey:keygenEC.privateKey publicKey:keygenEC.publicKey];
+}
 
-- (void)wrapUnwrapWithPrivateKey:(NSString *)privateKey publicKey:(NSString *)publicKey {
-    NSData *publicKeyData = [[NSData alloc] initWithBase64EncodedString:publicKey
-                                                                options:NSDataBase64DecodingIgnoreUnknownCharacters];
-    NSData *privateKeyData = [[NSData alloc] initWithBase64EncodedString:privateKey
-                                                                 options:NSDataBase64DecodingIgnoreUnknownCharacters];
 
+- (void)wrapUnwrapWithPrivateStringKey:(NSString *)privateKey publicStringKey:(NSString *)publicKey {
+  NSData *publicKeyData = [[NSData alloc] initWithBase64EncodedString:publicKey
+                                                              options:NSDataBase64DecodingIgnoreUnknownCharacters];
+  NSData *privateKeyData = [[NSData alloc] initWithBase64EncodedString:privateKey
+                                                               options:NSDataBase64DecodingIgnoreUnknownCharacters];
+  
+  return [self wrapUnwrapWithPrivateKey:privateKeyData publicKey:publicKeyData];
+}
+
+- (void)wrapUnwrapWithPrivateKey:(NSData *)privateKeyData publicKey:(NSData *)publicKeyData {
     TSMessage *encrypter = [[TSMessage alloc] initInEncryptModeWithPrivateKey:privateKeyData peerPublicKey:publicKeyData];
 
     NSString *message = @"- Knock, knock.\n- Who’s there?\n*very long pause...*\n- Java.";
 
 
     NSError *themisError = nil;
     NSData *encryptedMessage = [encrypter wrapData:[message dataUsingEncoding:NSUTF8StringEncoding] error:&themisError];
-    XCTAssertNil(themisError, @"wrapping data should be successful");
-    XCTAssertNotNil(encryptedMessage, @"wrapping data should return encrypted data");
-
+    XCTAssertNil(themisError, @"encrypting data should be successful");
+    XCTAssertNotNil(encryptedMessage, @"encrypting data should return encrypted data");
+  
     NSData *decryptedMessage = [encrypter unwrapData:encryptedMessage error:&themisError];
-    XCTAssertNil(themisError, @"unwrapping data should be successful");
-    XCTAssertNotNil(decryptedMessage, @"unwrapping data should return decrypted data");
+    XCTAssertNil(themisError, @"decrypting data should be successful");
+    XCTAssertNotNil(decryptedMessage, @"decrypting data should return decrypted data");
 
     NSString *resultString = [[NSString alloc] initWithData:decryptedMessage encoding:NSUTF8StringEncoding];
     XCTAssertTrue([message isEqualToString:resultString], @"initial string and decrypted string should be the same");
+  
+    // empty message
+    NSError *themisEmptyErrorEn = nil;
+    NSData *emEmptyMessageEn = [encrypter wrapData:NULL error:&themisEmptyErrorEn];
+    XCTAssertNotNil(themisEmptyErrorEn, @"encrypting data with empty message is error");
+    XCTAssertNil(emEmptyMessageEn, @"encrypting data with empty message should return nil message");
```

Comment:
In addition to testing `nil` keys and message we may also verify _empty_ ones.

Reply:
added